librz/il/definitions/vm_defines.h

```diff
@@ -0,0 +1,103 @@
+// SPDX-FileCopyrightText: 2021 heersin <teablearcher@gmail.com>
+// SPDX-License-Identifier: LGPL-3.0-only
+
+#ifndef CORE_THEORY_VM_DEFINITIONS_H
+#define CORE_THEORY_VM_DEFINITIONS_H
+
+#include "bitvector.h"
+#include "bool.h"
+#include "rz_lib.h"
+
+// ========= Basic Struct ============
+// RzIL_PURE is only a mark to remind the developer
+#define RzIL_PURE_VAL
+#define RzIL_BITV
+#define RzIL_BOOL
+#define RzIL_VAR
+#define RzIL_EFF
+#define RzIL_LABLE
+#define RzIL_MEM
+
+typedef enum {
+	RZIL_VAR_TYPE_BV,
+	RZIL_VAR_TYPE_BOOL,
+	RZIL_VAR_TYPE_UNK, // Unkown value
+} RZIL_VAR_TYPE;
+
+typedef union {
+	BitVector bv;
+	Bool b;
+} RzValUnion;
+
+/**
+ *  \struct rz_il_var_t
+ *  \brief structure of RzILVar
+ */
+struct rz_il_var_t {
+	string var_name; ///< name of variable
+	RZIL_VAR_TYPE type; ///< data type of variable
+};
+
+/**
+ *  \struct rz_il_val_t
+ *  \brief structure of RzILVal
+ */
+struct rz_il_val_t {
+	RZIL_VAR_TYPE type; ///< type of value
+	RzValUnion data; ///< data pointer
+};
+
+typedef void (*RzILBagFreeFunc)(void *);
+/**
+ *  \struct rz_il_bag_t
+ *  \brief structure of RzILBag, used to store RzILVal instances and manage them
```

Comment:
Please also emphasize here that the primary purpose is to clean up unused variables and free everything at the end, otherwise it rather suggests that the values are also actively fetched from the bag.